### PR TITLE
Added restore key documentation to pipeline caching

### PR DESCRIPTION
**Pull request by owenhuynMSFT**

### CODE REVIEW

File: docs/pipelines/caching/index.md [hunk 1]
```diff
@@ -88,6 +88,36 @@ In this example, the cache key contains three parts: a static string ("yarn"), t
...
+  YARN_CACHE_FOLDER: $(Pipeline.Workspace)/.yarn
+
+steps:
+- task: CacheBeta@0azure pipelines 
+  inputs:
+    key: yarn | $(Agent.OS) | yarn.lock
+    path: $(YARN_CACHE_FOLDER)
+    restoreKeys:
+      - yarn | $(Agent.OS)
+      - yarn
```
Comment (fadnavistanmay) [inline docs/pipelines/caching/index.md]: I think in yaml it should be:
restoreKeys: |
    yarn | $(Agent.OS)
    yarn

File: docs/pipelines/caching/index.md [hunk 2]
```diff
@@ -88,6 +88,36 @@ In this example, the cache key contains three parts: a static string ("yarn"), t
...
+#### Example
+
+Here is an example on how to use restore keys by Yarn:
+
+```yaml
+variables:
+  YARN_CACHE_FOLDER: $(Pipeline.Workspace)/.yarn
+
+steps:
+- task: CacheBeta@0azure pipelines
```
Comment (fadnavistanmay) [inline docs/pipelines/caching/index.md]: extra "azure pipelines" ?

File: docs/pipelines/caching/index.md (hunk 1 above)
Comment (johnterickson) [reply docs/pipelines/caching/index.md]: yeah i don't think there is a dash

File: docs/pipelines/caching/index.md [hunk 3]
```diff
@@ -88,6 +88,36 @@ In this example, the cache key contains three parts: a static string ("yarn"), t
 
 On the first run after the task is added, the cache step will report a "cache miss" since the cache identified by this key does not exist. After the last step, a cache will be created from the files in `$(Pipeline.Workspace)/.yarn` and uploaded. On the next run, the cache step will report a "cache hit" and the contents of the cache will be downloaded and restored.
 
+#### Restore keys
+
+`restoreKeys` can be used if one wants to query against multiple exact keys or key prefixes. This is used to fallback to another key in the case that a `key` does not yield a hit. A restore key will search for a key by prefix and yield the latest created cache entry as a result. This is useful if the pipeline is unable to find an exact match but wants to use a partial cache hit instead. To insert multiple restore keys, simply delimit them by using a new line and `-` character for indicate the restore key (see the example for more details). The order of which restore keys will be tried against will be from top to bottom.
```
Comment (elbatk) [inline docs/pipelines/caching/index.md]: "for indicate" --> "to indicate"

File: docs/pipelines/caching/index.md (hunk 1 above)
Comment (owenhuynMSFT) [reply docs/pipelines/caching/index.md]: Changed to new line delimiter

File: docs/pipelines/caching/index.md (hunk 2 above)
Comment (owenhuynMSFT) [reply docs/pipelines/caching/index.md]: Removed and changed for version 1

**Pull request merged**